lib/apiservers/engine/backends/volume.go

```diff
@@ -110,21 +109,24 @@ func (v *Volume) VolumeCreate(name, driverName string, opts, labels map[string]s
 		return result, derr.NewErrorWithStatusCode(fmt.Errorf("Bad Driver Arg: %s", varErr), http.StatusBadRequest)
 	}
 
-	// TODO: setup name randomization if name == nil
+	if model.Name == "" {
+		volumeUUID, _ := uuid.NewUUID()
+		model.Name = volumeUUID.String()
+	}
 
 	res, err := client.Storage.CreateVolume(storage.NewCreateVolumeParams().WithVolumeRequest(model))
 	if err != nil {
 		switch err := err.(type) {
 
 		case *storage.CreateVolumeInternalServerError:
 			// FIXME: right now this does not return an error model...
-			return result, derr.NewErrorWithStatusCode(fmt.Errorf("Server error from Portlayer: %s", err.Error()), http.StatusInternalServerError)
+			return result, derr.NewErrorWithStatusCode(fmt.Errorf("%s", err.Error()), http.StatusInternalServerError)
 
 		case *storage.CreateVolumeDefault:
-			return result, derr.NewErrorWithStatusCode(fmt.Errorf("Server error from Portlayer: %s", err.Payload.Message), http.StatusInternalServerError)
+			return result, derr.NewErrorWithStatusCode(fmt.Errorf("%s", err.Payload.Message), http.StatusInternalServerError)
```

Comment:
Unfortunately according to api v1.23 we the volume api only returns a `201` for `no error` and a `500` for `error from daemon server` and the 500 return string can change from time to time since it is meant as a custom return i believe. I will take a look at the what docker returns, it is likely we will have different messages since we are only dealing with block devices.